pkg/clusterversion/cv.go

```diff
@@ -159,16 +174,34 @@ func isEqualVersion(cv *configv1.ClusterVersion, uc *upgradev1alpha1.UpgradeConf
 	return false
 }
 
-// hasUpgradeCommenced checks if the upgrade has commenced
+// isEqualImage compare the upgrade version state for cv and uc
+func isEqualImage(cv *configv1.ClusterVersion, uc *upgradev1alpha1.UpgradeConfig) bool {
+	if cv.Spec.DesiredUpdate != nil &&
+		cv.Spec.DesiredUpdate.Image == uc.Spec.Desired.Image {
+		return true
+	}
+
+	return false
+}
+
+// hasUpgradeCommenced checks if the upgrade has commenced based on version or image
 func (c *clusterVersionClient) HasUpgradeCommenced(uc *upgradev1alpha1.UpgradeConfig) (bool, error) {
 
 	clusterVersion, err := c.GetClusterVersion()
 	if err != nil {
 		return false, err
 	}
 
-	if !isEqualVersion(clusterVersion, uc) {
-		return false, nil
+	if len(uc.Spec.Desired.Version) > 0 && len(uc.Spec.Desired.Channel) > 0 {
+		if !isEqualVersion(clusterVersion, uc) {
+			return false, nil
+		}
+	}
+
+	if len(uc.Spec.Desired.Version) > 0 && len(uc.Spec.Desired.Image) > 0 {
```

Comment:
As above , would it be more appropriate to check these attributes with `!= "" ?`